### ISSUE TITLE
Is there any way to run this anymore with Python2.7 no longer being supported?

### ISSUE THREAD

**Issue by samgermain** [ksallberg/wallstreet_tycoon#2]
I'm trying to create a [python2 virtualenv][1], so I try `pip install virtualenv` and get

```
Traceback (most recent call last):
  File "/usr/local/bin/pip", line 11, in <module>
    load_entry_point('pip==21.1.1', 'console_scripts', 'pip')()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 2843, in load_entry_point
    return ep.load()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 2434, in load
    return self.resolve()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 2440, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Library/Python/2.7/site-packages/pip-21.1.1-py2.7.egg/pip/__init__.py", line 1, in <module>
    from typing import List, Optional
ImportError: No module named typing
```

I get an almost identical error with `pip install typing`

    Traceback (most recent call last):
      File "/usr/local/bin/pip", line 11, in <module>
        load_entry_point('pip==21.1.1', 'console_scripts', 'pip')()
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 489, in load_entry_point
        return get_distribution(dist).load_entry_point(group, name)
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 2843, in load_entry_point
        return ep.load()
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 2434, in load
        return self.resolve()
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 2440, in resolve
        module = __import__(self.module_name, fromlist=['__name__'], level=0)
      File "/Library/Python/2.7/site-packages/pip-21.1.1-py2.7.egg/pip/__init__.py", line 1, in <module>
        from typing import List, Optional
    ImportError: No module named typing

  [1]: https://gist.github.com/simonw/4835a22c79a8d3c29dd155c716b19e16
  [2]: https://github.com/uber/uber-poet/issues/30#issuecomment-516216406

**Comment by ksallberg** [comment]
:)

Not sure. I tried doing `pip2 install typing`, and it seemed to work. But then I ran into `error loading helperspygletpyglet 2.0.dev3 requires Python 3.5 or newer.` when trying to run the game. Could be resolvable, but could also be the case that you will run into more and more of these kind of issues, for each issue you resolve.

The problem is, younger version of me thought it was a good idea to bring in both pygame, django and an ORM to speed up completion of this course work, but now this little project is stuck with a lot of dependencies, and I don't have the time or motivation to keep it up to date. Maybe I should just archive it to stop people from getting disappointed in this no longer working project?

Either way, the python code could of course be ported to python3, or it could be executed on a system that has been frozen in time maybe 8 years ago (like a VM) where python2.7 was still what people used.

**Comment by samgermain** [comment]
Yeah I'm trying to run it on a vm, I'm trying to change the version dependencies to == instead of >= like

```
      install_requires = ['Django == 1.5.1',
                          'pyglet === 1.1.4',
                          'tiledtmxloader == 3.0.3',
                          'pygame==1.9.1',
                         ],
```
                         
but am getting bugs like `No local packages or working download links found for pygame==1.9.1` or `ERROR: Could not find a version that satisfies the requirement tiledtmxloader==3.0.3 (from versions: 3.1.0.131)`
                         
Do they just make older versions of things not available?

I was able to install pyglet 1.1.4, but I don't know if the other packages will let me, which seems kinda shitty, I feel like there should be a way to install what you need just to execute a program

**Comment by samgermain** [comment]
[I'm working on figuring it out](https://stackoverflow.com/questions/67658545/installing-old-versions-of-dependencies-within-a-python-virtualenv-and-receiving)

I'm hoping once I get it running, that I can insert my own candlestick data into it (here's some [example data](https://github.com/samgermain/binance_usdt) ), but speed it up to like 60x so that I can practice on the current market really quickly. Do you have any idea if this will be possible?

**Comment by ksallberg** [comment]
Hm, this project is mostly a 2D isometric view game where you have a character that can walk around town and enter buildings etc, and you see some bots walk around too. Then a bunch of GUI elements. I wanted to do some stock simulation but there wasn't much time for that in the course work, we had to focus on GUI, animation, etc. There is no market aspect to this really. Prices on "stocks" go up and down (randomly) according to a super simple formula. I think it would be better to create something from scratch or find another project.

This is what data variables this project uses for "stocks" and "companies":

https://github.com/ksallberg/wallstreet_tycoon/blob/master/gamemodels/models.py

Here is how prices are calculated:

https://github.com/ksallberg/wallstreet_tycoon/blob/master/logic/priceManipulation.py

Rest of the project is really just generation of fake company names, and using pygame for rendering characters, houses, streets etc.

**Comment by ksallberg** [comment]
I did try this maybe 10 years ago, https://www.eurogamer.net/articles/wst_p looks really dated now and I don't know if you can import data.

Then apparently you have https://store.steampowered.com/app/985710/Wall_Street_Tycoon/ which is a newer game, albeit with poor ratings.

**Comment by samgermain** [comment]
I got this working, I had to change the install_requires into == instead of >=, I used python==1.9.3 instead of 1.9.1, there were some problems with 1.9.1

```
install_requires = ['Django == 1.5.1',
                          'pyglet == 1.1.4',
                          'tiledtmxloader == 3.0.3.114',
                          'pygame==1.9.3',
                         ],
      
     )
 ```
     
I also had to download the `pygame` and `tiledtmxloader` from source

pip install https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/pytmxloader/tiledtmxloader-3.0.3.114.zip
pip install https://pypi.python.org/packages/61/06/3c25051549c252cc6fde01c8aeae90b96831370884504fe428a623316def/pygame-1.9.3.tar.gz#md5=ac744ea6952b68d5f2b6d02a6d8e836c

--------------------------

I had to do this all within a python2 virtualenv which is talked about on this [Stack Overflow question](https://stackoverflow.com/a/67656933/6331353). Make sure you read the comments if you do this, the answer alone didn't do it.

**Comment by ksallberg** [comment]
good work, so you could actually see the graphics? :P

**Comment by samgermain** [comment]
I got to the startup screen, but then I read your comment about what the game was about and didn't go any further than that. The instructions are there now though if anyone else wants to play the game
